lib/modules/manager/pixi/schema.ts

```diff
@@ -1,10 +1,207 @@
+import is from '@sindresorhus/is';
 import { z } from 'zod';
-import { Toml, Yaml } from '../../../util/schema-utils';
+import { isNotNullOrUndefined } from '../../../util/array';
+import { LooseRecord, Toml, Yaml } from '../../../util/schema-utils';
+import { CondaDatasource } from '../../datasource/conda/';
+import { PypiDatasource } from '../../datasource/pypi';
+import * as condaVersion from '../../versioning/conda/';
+import { id as gitRefVersionID } from '../../versioning/git';
+import { id as pep440VersionID } from '../../versioning/pep440/';
+import type { PackageDependency } from '../types';
+
+export interface PixiPackageDependency extends PackageDependency {
+  channel?: string;
+  channels?: (string | { channel: string; priority: number })[];
+}
+
+function collectNamedPackages(
+  packages: Record<string, PackageDependency | null>,
+): PackageDependency[] {
+  return Object.entries(packages)
+    .map(([depName, config]) => {
+      if (is.nullOrUndefined(config)) {
+        return;
+      }
+
+      return {
+        ...config,
+        depName,
+      };
+    })
+    .filter((dep) => isNotNullOrUndefined(dep));
+}
+
+const pypiDependencies = z
+  .record(
+    z.string(),
+    z.union([
+      z.string().transform((version) => {
+        return {
+          currentValue: version,
+          versioning: pep440VersionID,
+          datasource: PypiDatasource.id,
+          depType: 'pypi-dependencies',
+        } satisfies PixiPackageDependency;
+      }),
+      z.object({ version: z.string() }).transform(({ version }) => {
```

Comment:
Only use top level transforms if you are modifying files. https://github.com/renovatebot/renovate/blob/main/docs/development/zod.md#rename-and-move-fields-at-the-top-level-transform